intelligence/docs/source/index.rst

```diff
@@ -69,168 +127,276 @@ Once your instance is set up, you can start chatting! Here’s a basic example:
-                });
-                if (reply.ok){
-                  console.log("Full response:", reply.message.content);
-                }
+            import { ChatResponseResult, FlowerIntelligence, type StreamEvent } from '@flwr/flwr';
 
-        .. tab-item:: JavaScript
-            :sync: js
+            // Access the singleton instance
+            const fi: FlowerIntelligence = FlowerIntelligence.instance;
 
-            .. code-block:: js
+            async function main() {
+              const response: ChatResponseResult = await fi.chat('Why is the sky blue?', {
+                model: 'meta/llama3.2-1b/instruct-fp16',
+                stream: true,
+                onStreamEvent: (event: StreamEvent) => console.log(event.chunk)
+              });
+
+              if (!response.ok) {
+                console.error(`${response.failure.code}: ${response.failure.description}`);
+              } else {
+                console.log('Full response:', response.message.content);
+              }
+            }
 
-                const reply = await fi.chat("Why is the sky blue?", {
-                  stream: true,
-                  onStreamEvent: (event) => console.log(event.chunk)
-                });
-                console.log("Full response:", reply.message.content);
+            await main().then().catch();
 
-2. **Using Roles**
+    .. tab-item:: JavaScript
+        :sync: js
 
-   Provide an array of messages to maintain conversation context. Instead of
-   just passing a string to the ``chat`` method, you can use ``Message`` or an array
-   of ``Message``. This lets you use different roles like system messages:
+        .. code-block:: js
+
+            import { FlowerIntelligence } from '@flwr/flwr';
+
+            // Access the singleton instance
+            const fi = FlowerIntelligence.instance;
+
+            async function main() {
+              const response = await fi.chat('Why is the sky blue?', {
+                model: 'meta/llama3.2-1b/instruct-fp16',
+                stream: true,
+                onStreamEvent: (event) => console.log(event.chunk)
+              });
+
+              if (!response.ok) {
+                console.error(`${response.failure.code}: ${response.failure.description}`);
+              } else {
+                console.log(response.message.content);
+              }
+            }
 
-    .. tab-set::
-        :sync-group: category
+            await main().then().catch();
 
-        .. tab-item:: TypeScript
-            :sync: ts
+Use Roles
+---------
+
+To enhance conversation context, Flower Intelligence supports the use of different message roles. Instead of simply sending a single string, you can provide an array of :doc:`messages <ts-api-ref/interfaces/Message>` with designated roles such as “system” and “user.” This allows you to define the behavior and context of the conversation more clearly, ensuring that the assistant responds in a way that’s tailored to the scenario.
```

Comment:
```suggestion
To enhance conversation context, Flower Intelligence supports the use of different message roles. Instead of simply sending a single string, you can provide an array of :doc:`messages <ts-api-ref/interfaces/Message>` with designated roles such as ``system`` and ``user``. This allows you to define the behavior and context of the conversation more clearly, ensuring that the assistant responds in a way that’s tailored to the scenario.
```